Source/Dafny/Resolver.cs

```diff
@@ -12349,7 +12349,11 @@ private class RBranchExpr : RBranch {
 
     private void CheckLinearVarPattern(Type type, IdPattern pat, ResolveOpts opts) {
       if (pat.Arguments != null) {
-        reporter.Error(MessageSource.Resolver, pat.Tok, "member {0} does not exist in type {1}", pat.Id, type);
+        if (pat.Id == BuiltIns.TupleTypeCtorNamePrefix + "1") {
```

Comment:
Could you assign a variable to `BuiltIns.TupleTypeCtorNamePrefix + "1"`, like `BuiltIns.SingletonTupleName` ?

Reply:
A similar construction was used in several places, so I added a helper method `BuiltIns.TupleTypeCtorName(int dims)`.